src/marqo/tensor_search/on_start_script.py

```diff
@@ -84,8 +84,17 @@ def id_to_device(id):
             if id < 0:
                 return ['cpu']
             return [torch.cuda.get_device_name(id)]
-        
-        device_count = 0 if not torch.cuda.is_available() else torch.cuda.device_count()
+
+        """
+            This is set once at startup time. We assume it will NOT change,
+            if it does, health check should throw a warning.
+        """
+        device_count = 0
+        os.environ["MARQO_BEST_AVAILABLE_DEVICE"] = "cpu"
+        if torch.cuda.is_available():
+            device_count = torch.cuda.device_count()
+            os.environ["MARQO_BEST_AVAILABLE_DEVICE"] = "cuda"
```

Comment:
it then becomes easier to create unit tests for it

Reply:
Can do! Although I kept it in `CUDAAvailable` to reuse the `if torch.cuda.is_available()` call. It's fine tbh since it only happens ocne on startup anyway